lib/botmetrics.rb

```diff
@@ -30,6 +30,20 @@ def register_bot!(token, opts = {})
       response.code == 201
     end
 
+    def message(team_id:, channel: nil, user: nil, text: nil, attachments: nil)
+      params = {
+        "message[team_id]"     => team_id,
+        "message[channel]"     => channel,
+        "message[user]"        => user,
+        "message[text]"        => text,
+        "message[attachments]" => attachments
+      }.delete_if { |_, v| v.nil? }
+
+      response = HTTP.auth(api_key).post("#{api_url}/messages", params: params)
```

Comment:
the server will handle it but we should raise an error if both `message[channel]` and `message[user]` are blank. Similarly if both `message[text]` and `message[attachments]` are blank.